src/Agent.Worker/ExecutionContext.cs

```diff
@@ -212,6 +226,18 @@ public void Start(string currentOperation = null)
             _record.State = TimelineRecordState.InProgress;
 
             _jobServerQueue.QueueTimelineRecordUpdate(_mainTimelineId, _record);
+
+            if (_localLogs)
+            {
+                var localLogsJobFolder = Path.Combine(_localLogsFolderPath, _mainTimelineId.ToString());
+                Directory.CreateDirectory(localLogsJobFolder);
+
+                _localLogsFile = Path.Combine(localLogsJobFolder, $"{_record.Name}-{Guid.NewGuid().ToString()}.log");
```

Comment:
The TimelineRecord has its own Guid Id that correlates with all the server side state. How about using that in the log name rather than a new Guid?

Reply:
Is that unique for each job?  The folders come in as jobs.

Reply:
Yeah, there is a TimelineRecord for each job. That is what stores the status, start, finish times, etc.